src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,149 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+
+import duke.data.Storage;
+import duke.data.TaskList;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+import duke.user.DukeException;
+import duke.user.Parser;
+import duke.user.Ui;
+
+/**
+ * Main class for the Duke program.
+ */
+public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+
+    /**
+     * Constructor for the Duke instance.
+     *
+     * @param filePath the String representing the path of the file where the data is saved
+     */
+    public Duke(String filePath) {
+        storage = new Storage(filePath);
+        tasks = new TaskList(storage.loadFromFile());
+        ui = new Ui();
+        assert storage != null;
+    }
+
+    /**
+     * Default constructor for Duke.
+     * Required for JavaFX GUI
+     */
+    public Duke() {
+        this("data/Duke.txt");
+    }
+
+    /**
+     * Handles the user input and sends it to the parser to perform the commands, if valid.
+     */
+    public String getResponse(String input) {
+
+        Parser parser = new Parser(tasks);
+        boolean taskListIsUpdated = false;
+        boolean taskListIsAddedTo = false;
```

Comment:
For boolean variables it is better to start with is or has. Maybe we can consider changing this name to hasAddedTaskList? I have no difficulty understanding your naming, but maybe it is better to start with is or has.